src/features/defi/providers/osmosis/components/OsmosisManager/Withdraw/OsmosisWithdraw.tsx

```diff
@@ -0,0 +1,181 @@
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+import type { DefiStepProps } from 'components/DeFi/components/Steps'
+import { Steps } from 'components/DeFi/components/Steps'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { marketData } from 'state/slices/marketDataSlice/marketDataSlice'
+import type { OsmosisPool } from 'state/slices/opportunitiesSlice/resolvers/osmosis/utils'
+import {
+  getPool,
+  getPoolIdFromAssetReference,
+} from 'state/slices/opportunitiesSlice/resolvers/osmosis/utils'
+import type { LpId } from 'state/slices/opportunitiesSlice/types'
+import { toOpportunityId } from 'state/slices/opportunitiesSlice/utils'
+import {
+  selectAssetById,
+  selectEarnUserLpOpportunity,
+  selectPortfolioLoading,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { Confirm } from './components/Confirm'
+import { Status } from './components/Status'
+import { Withdraw } from './components/Withdraw'
+import { OsmosisWithdrawActionType } from './WithdrawCommon'
+import { WithdrawContext } from './WithdrawContext'
+import { initialState, reducer } from './WithdrawReducer'
+
+type OsmosisWithdrawProps = {
+  onAccountIdChange: AccountDropdownProps['onChange']
+  accountId: AccountId | undefined
+}
+
+export const OsmosisWithdraw: React.FC<OsmosisWithdrawProps> = ({
+  accountId,
+  onAccountIdChange: handleAccountIdChange,
+}) => {
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const translate = useTranslate()
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, assetNamespace, assetReference } = query
+  const userAddress = useMemo(() => accountId && fromAccountId(accountId).account, [accountId])
+
+  // Asset info
+  const assetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference,
+  })
+
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const loading = useSelector(selectPortfolioLoading)
+  if (!asset) throw new Error(`Asset not found for AssetId ${assetId}`)
+
+  const opportunityId: LpId | undefined = useMemo(
+    () => (assetId ? toOpportunityId({ chainId, assetNamespace, assetReference }) : undefined),
+    [assetId, assetNamespace, assetReference, chainId],
+  )
+
+  const OsmosisLpOpportunityFilter = useMemo(
+    () => ({
+      lpId: opportunityId,
+      assetId,
+      accountId,
+    }),
+    [accountId, assetId, opportunityId],
+  )
+  const opportunity = useAppSelector(state =>
+    selectEarnUserLpOpportunity(state, OsmosisLpOpportunityFilter),
+  )
+
+  const underlyingAsset0Id = useMemo(
+    () => opportunity?.underlyingAssetIds[0] ?? '',
+    [opportunity?.underlyingAssetIds],
+  )
+  const underlyingAsset1Id = useMemo(
+    () => opportunity?.underlyingAssetIds[1] ?? '',
+    [opportunity?.underlyingAssetIds],
+  )
```

Comment:
Can be passed as selector args directly with the same referential guarantees, but also fine like this, your call